app/queries/create_submission_mutator.rb

```diff
@@ -54,6 +54,52 @@ def create_submission
 
   private
 
+  # rubocop: disable Metrics/CyclomaticComplexity
+  def valid_response
+    return if checklist.respond_to?(:all?) && checklist.all? do |item|
+      item['title'].is_a?(String) && item['kind'].in?(Target.valid_checklist_kind_types) &&
+        item['status'].in?(TimelineEvent::CHECKLIST_STATUS_NO_ANSWER) && item['result'].is_a?(String) &&
+        valid_result(item['kind'], item['result'])
+    end
+
+    errors[:base] << 'Submission checklist is not valid.'
+  end
+
+  def valid_result(kind, result)
+    case kind
+      when Target::CHECKLIST_KIND_ATTACH_FILES
+        timeline_event_files.present?
+      when Target::CHECKLIST_KIND_ATTACH_LINKS
+        result.length >= 3 && result.length <= 2048
+      when Target::CHECKLIST_KIND_LONG_TEXT
+        result.length >= 1
+      when Target::CHECKLIST_KIND_MULTI_CHOICE
+        result.length >= 1
+      when Target::CHECKLIST_KIND_SHORT_TEXT
+        result.length >= 1
+      else
+        false
+    end
+  end
+
+  # rubocop: enable Metrics/CyclomaticComplexity
+
+  def attempted_minimum_questions
+    target_checklist.each do |c|
+      next if c['optional'] == true
+
+      item = checklist.select { |i| i["title"] == c['title'] }
+
+      next if item.present? && item.count == 1 && item.first['result'].is_a?(String)
+
+      errors[:base] << "Missing answer for question: #{c['title']}"
+    end
+  end
+
+  def target_checklist
+    @target_checklist ||= target.checklist
+  end
```

Comment:
This memoization doesn't achieve anything; `checklist` is a column on the `targets` table - its data has already been loaded into the `target` object.